### PR TITLE
[swift-parse-test] Tweak for review

### PR DESCRIPTION
**Pull request by rintaro**
Tweak fore the review on #69470 

* 'iteration' -> 'iterations'
* Comment to clarify cpu time vs wall clock time
* Return 'Error' from the executors

### CODE REVIEW

File: lib/DriverTool/swift_parse_test_main.cpp [hunk 1]
```diff
@@ -178,13 +194,16 @@ perform(const SmallVectorImpl<std::unique_ptr<llvm::MemoryBuffer>> &buffers,
...
       std::chrono::duration_cast<std::chrono::milliseconds>(cDuration).count();
 
+  // Throughputs are based on CPU time.
   auto byteTPS = totalBytes * duration_t::period::den / cDuration.count();
   auto lineTPS = totalLines * duration_t::period::den / cDuration.count();
 
   llvm::outs() << llvm::format("wall clock time (ms): %8d\n", tDisplay)
                << llvm::format("cpu time (ms):        %8d\n", cDisplay)
                << llvm::format("throughput (byte/s):  %8d\n", byteTPS)
                << llvm::format("throughput (line/s):  %8d\n", lineTPS);
```
Comment (rintaro) [inline lib/DriverTool/swift_parse_test_main.cpp:204]: I didn't change this because I didn't want to make these headers long.

**Comment by rintaro** [comment]
https://github.com/apple/swift/pull/69470#discussion_r1376777993
> Do you think we should extend this tool to do more parser tests? If it should just stay a performance measuring tool, I would name it swift-parser-performance-test.

I am not certain at this point. Maybe we can use this as a ASTGen test utility to verify the ASTGen result, or testing `SwiftIfConfig` combined with `CompilerInvocation`, etc.

> Also, I just want to note that swift-syntax has swift-parser-test (with an r). I don’t know if it’s good that this is spelled without an r (means we don’t have two utilities with the same name) or bad (inconsistency).

I don't really have strong opinion on this 😅

**Review by ahoppen — APPROVED** [review]
Thanks

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test Windows

File: lib/DriverTool/swift_parse_test_main.cpp [hunk 2]
```diff
@@ -87,19 +89,26 @@ struct LibParseExecutor {
...
 #if SWIFT_BUILD_SWIFT_SYNTAX
     // TODO: Implement 'ExecuteOptionFlag::SkipBodies'
     auto sourceFile = swift_ASTGen_parseSourceFile(
         buffer.getBufferStart(), buffer.getBufferSize(), /*moduleName=*/"",
         buffer.getBufferIdentifier().data(), /*ASTContext=*/nullptr);
     swift_ASTGen_destroySourceFile(sourceFile);
+    return llvm::Error::success();
+#else
+    return llvm::createStringError(llvm::inconvertibleErrorCode(),
+                                   "SwiftParser is not supported")
```
Comment (finagolfin) [inline lib/DriverTool/swift_parse_test_main.cpp:111]: @rintaro, this C++ syntax error [broke builds that don't use SwiftSyntax](https://ci-external.swift.org/job/oss-swift-RA-linux-ubuntu-16.04-android-arm64/5711/console):
```
/home/ubuntu/jenkins/workspace/oss-swift-RA-linux-ubuntu-16.04-android-arm64/swift/lib/DriverTool/swift_parse_test_main.cpp:111:67: error: expected ';' after return statement
                                   "SwiftParser is not supported")
```

**Pull request merged**